project-docs/javascript-linter-spec.md

```diff
@@ -0,0 +1,163 @@
+- The `<dl>` must contain a single `<dt><dd>` pair.
+- The `<dt>` must contain either:
+  - only a single `<code>` element, that contains only a single `<a>` element
+  - only a call to the `{{jsxref}}` macro.
+
+#### data.constructor_properties?
+
+This is an optional ingredient. If a page contains a section demarcated by `H2#Constructor_properties`, then it must meet the following requirements:
+
+- Must contain only a `<dl>` element.
+- The `<dl>` must contain one or more `<dt><dd>` pairs.
+- The `<dt>` must contain either:
+  - only a single `<code>` element, that contains only a single `<a>` element
+  - only a call to the `{{jsxref}}` macro.
+- `<dt><dd>` pairs must be ordered by the alphabetical order of the `<dt><code><a>` text content, or by the display name given in the `{{jsxref}}` macro call.
+
+#### data.static_properties?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Static_properties`.
+
+#### data.static_methods?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Static_methods`.
+
+#### data.instance_properties?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Instance_properties`.
+
+#### data.instance_methods?
+
+This is an optional ingredient. It's just the same as `data.constructor_properties?`, except the section is demarcated by `H2#Instance_methods`.
+
+#### data.examples
+
+To satisfy this ingredient a page must have a section demarcated by `H2#Examples`. It must contain one or more sections demarcated by an H3. Each of these sections represents a single example.
+
+Each of these examples may include "code sections". These are marked up as follows:
+
+- HTML code section: `<pre class="brush: html language-html"><code class=" language-html">`
+- CSS code section: `<pre class="brush: css language-css"><code class=" language-css">`
+- JavaScript code section: `<pre class="brush: js language-js"><code class=" language-js">`
+
+Examples must follow some rules about which sections may appear and how they are ordered:
+
+- An example may contain zero or one of each type of section.
+- If an example includes a JavaScript code section, it must be the last element in the section.
+- If an example includes a CSS code section, it may only be followed by a JavaScript section.
+- If an example includes an HTML code section, it may only be followed by a CSS section.
+
+Another way to put this is: each example is structured like:
+
+```
+unstructured prose
+optional HTML section
+optional CSS section
+optional JavaScript section
+```
+
+#### data.specifications
+
+To satisfy this ingredient a page must have a section demarcated by `H2#Specifications`. It must contain either a call to the `{{SpecName}}` macro, or a text node containing the text "Not part of any standard".
```

Comment:
I see this is without the full stop in the linter code, too, but should it be required?

```suggestion
To satisfy this ingredient a page must have a section demarcated by `H2#Specifications`. It must contain either a call to the `{{SpecName}}` macro, or a text node containing the text "Not part of any standard.".
```